File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        var ones = 0;
        for(var num : s.toCharArray())
            if(num == '1') ones++;
        
        int maxScore = 0, currZero = 0, currOne = 0;
        for(var i=0; i<s.length() - 1; i++){
            if(s.charAt(i) == '0') currZero++;
            else currOne++;

            var currScore = currZero + ones - currOne;
            maxScore = Math.max(maxScore, currScore);
        }
        
        return maxScore;
    }
}

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones = 0
        for num in s:
            ones += 1 if num == '1' else 0
        
        max_score = curr_zero = curr_one = 0
        for i in range(len(s)-1):
            if s[i] == '0': curr_zero += 1
            else: curr_one += 1

            curr_score = curr_zero + ones - curr_one
            max_score = max(max_score, curr_score)
        
        return max_score
